fades/pipmanager.py

```diff
@@ -87,16 +90,22 @@ def get_version(self, dependency):
                          'Run with -v or check the logs for details')
             return ''
 
-    def _brute_force_install_pip(self):
+    def download_pip_installer(self):
+        u = request.urlopen(PIP_INSTALLER)
+        with contextlib.closing(u), \
+                tempfile.NamedTemporaryFile('wb') as f:
+            shutil.copyfileobj(u, f)
+            f.flush()
+            shutil.copy(f.name, self.pip_installer_fname)
+
```

Comment:
"copy" is not an atomic operation, you need to use os.rename() here.

For the rename to really work, you need the file in the same partition, so you have two choices here: or constrain the NamedTemporaryFile call for that, or just use something like self.pip_installer_fname + ".temp"